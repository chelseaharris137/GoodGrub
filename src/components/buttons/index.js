export * from './Button';
export * from './LinkedButton';
