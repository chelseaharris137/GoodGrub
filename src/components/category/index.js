export * from './Categories';
export * from './CategoryItem';
