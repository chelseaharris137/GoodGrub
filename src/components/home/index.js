export * from './About';
export * from './HeroSection';
