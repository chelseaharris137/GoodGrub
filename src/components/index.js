export * from './buttons/Button';
export * from './buttons/LinkedButton';
export * from './category/Categories';
export * from './category/CategoryItem';
export * from './home/About';
export * from './home/HeroSection';
export * from './layout/Footer';
export * from './layout/Layout';
export * from './layout/Navbar';
export * from './mealsPage/IngredientsTable';
export * from './mealsPage/SearchBar';
export * from './mealsPage/SingleMealCard';
export * from './text/PointerText';
export * from './text/Text';
export * from './text/Title';
