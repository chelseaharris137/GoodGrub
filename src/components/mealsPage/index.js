export * from './IngredientsTable';
export * from './SearchBar';
export * from './SingleMealCard';
