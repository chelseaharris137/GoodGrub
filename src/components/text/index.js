export * from './PointerText';
export * from './Text';
export * from './Title';
