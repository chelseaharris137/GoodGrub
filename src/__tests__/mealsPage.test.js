import React from 'react';
import { render } from '@testing-library/react';
import { IngredientsTable, SearchBar, SingleMealCard } from '@/components';

describe('`IngredientsTable` component', () => {
  it('Does `IngredientsTable` match DOM Snapshot', () => {
    const props = {
      ingredientsWithMeasures: [
        { index: 1, ingredient: 'flour', measure: '1 cup' },
        { index: 2, ingredient: 'sugar', measure: '1/2 cup' },
        { index: 3, ingredient: 'vanilla', measure: '1/2 teaspoon' },
      ],
    };

    const ingredientsTable = render(<IngredientsTable {...props} />);

    expect(ingredientsTable).toMatchSnapshot();
  });
});

describe('`SearchBar` component', () => {
  it('Does `SearchBar` match DOM Snapshot', () => {
    const props = {
      searchText: 'hello world',
      setSearchText: () => {},
    };

    const searchBar = render(<SearchBar {...props} />);

    expect(searchBar).toMatchSnapshot();
  });
});

describe('`SingleMealCard` component', () => {
  it('Does `SingleMealCard` match DOM Snapshot', () => {
    const props = {
      meal: {
        idMeal: '45667',
        strMeal: 'BBQ Chicken & Rice',
        strMealThumb: '/image.jpg',
      },
    };

    const singleMealCard = render(<SingleMealCard {...props} />);

    expect(singleMealCard).toMatchSnapshot();
  });
});
