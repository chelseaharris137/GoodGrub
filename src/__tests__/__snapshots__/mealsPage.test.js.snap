// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`IngredientsTable\` component Does \`IngredientsTable\` match DOM Snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h2
        class=""
      >
        Ingredients
      </h2>
      <table>
        <tbody>
          <tr>
            <td>
              <p
                class=""
              >
                flour
              </p>
            </td>
            <td>
              <p
                class=""
              >
                1 cup
              </p>
            </td>
          </tr>
          <tr>
            <td>
              <p
                class=""
              >
                sugar
              </p>
            </td>
            <td>
              <p
                class=""
              >
                1/2 cup
              </p>
            </td>
          </tr>
          <tr>
            <td>
              <p
                class=""
              >
                vanilla
              </p>
            </td>
            <td>
              <p
                class=""
              >
                1/2 teaspoon
              </p>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </body>,
  "container": <div>
    <h2
      class=""
    >
      Ingredients
    </h2>
    <table>
      <tbody>
        <tr>
          <td>
            <p
              class=""
            >
              flour
            </p>
          </td>
          <td>
            <p
              class=""
            >
              1 cup
            </p>
          </td>
        </tr>
        <tr>
          <td>
            <p
              class=""
            >
              sugar
            </p>
          </td>
          <td>
            <p
              class=""
            >
              1/2 cup
            </p>
          </td>
        </tr>
        <tr>
          <td>
            <p
              class=""
            >
              vanilla
            </p>
          </td>
          <td>
            <p
              class=""
            >
              1/2 teaspoon
            </p>
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`\`SearchBar\` component Does \`SearchBar\` match DOM Snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <input
        value="hello world"
      />
    </div>
  </body>,
  "container": <div>
    <input
      value="hello world"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`\`SingleMealCard\` component Does \`SingleMealCard\` match DOM Snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        href="/meals/45667"
      >
        <img
          alt="BBQ Chicken & Rice"
          data-nimg="1"
          decoding="async"
          height="200"
          loading="lazy"
          src="/_next/image?url=%2Fimage.jpg&w=640&q=75"
          srcset="/_next/image?url=%2Fimage.jpg&w=384&q=75 1x, /_next/image?url=%2Fimage.jpg&w=640&q=75 2x"
          style="color: transparent;"
          width="290"
        />
        <h2
          class=""
        >
          BBQ Chicken & Rice
        </h2>
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      href="/meals/45667"
    >
      <img
        alt="BBQ Chicken & Rice"
        data-nimg="1"
        decoding="async"
        height="200"
        loading="lazy"
        src="/_next/image?url=%2Fimage.jpg&w=640&q=75"
        srcset="/_next/image?url=%2Fimage.jpg&w=384&q=75 1x, /_next/image?url=%2Fimage.jpg&w=640&q=75 2x"
        style="color: transparent;"
        width="290"
      />
      <h2
        class=""
      >
        BBQ Chicken & Rice
      </h2>
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
